backend/algorithms/objects/conditions.py

```diff
@@ -51,6 +55,18 @@ def is_path_to(self, course: str) -> bool:
         """ checks if 'course' is able to meet any subtree's requirements"""
         pass
 
+    @abstractmethod
+    def condition_to_model(self, model: cp_model.CpModel, user: User, courses: list[Tuple[cp_model.IntVar, Course]], course_variable: cp_model.IntVar) -> list[cp_model.Constraint]:
+        # just add straight up true or false. This default implementation works for things based only on the user, and not their courses
+        is_valid, _ = self.validate(user)
+        return [model.AddBoolAnd(is_valid)]
+
+    @abstractmethod
+    def condition_negation(self, model: cp_model.CpModel, user: User, courses: list[Tuple[cp_model.IntVar, Course]], course_variable: cp_model.IntVar) -> list[cp_model.Constraint]:
```

Comment:
this negation is needed only for the OR condition to work. We use it to make a condition on a boolean that is tied to it, then we can say that any of them should be true as a condition.